animations/hero-animations/index.md

```diff
@@ -0,0 +1,670 @@
+</li>&nbsp;
+
+<li markdown="1">Radial hero animations<br>
+In _radial hero animation_, as the hero flies between routes
+its shape appears to change from circular to rectangular.
+
+The following video (recorded at slow speed),
+shows an example of a radial hero animation. At the start,
+a row of three images appears at the bottom of the route.
+Tapping any of the circles flies that image to a new route
+that displays it with a square shape.
+Tapping the square image flies the hero back to
+the original route, displayed with a circular shape.
+
+<center><iframe width="560" height="315" src="https://www.youtube.com/embed/V735TEHW5UI" frameborder="0" allowfullscreen></iframe></center>
+</li>
+</ul>
+
+Before moving to the sections specific to
+[standard](#standard-hero-animations)
+or [radial](#radial-hero-animations) hero animations,
+read [basic structure of a hero animation](#basic-structure)
+to learn how to structure hero animation code,
+and [behind the scenes](#behind-the-scenes) to understand
+how Flutter performs a hero animation.
+
+<a name="basic-structure"></a>
+## Basic structure of a hero animation
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* Use two hero widgets with matching tags to implement the transition.
+* The Navigator manages a stack containing the app’s routes.
+* Pushing a route on or popping a route from the Navigator’s stack
+  triggers the transition.
+* The Flutter framework calculates a rectangle tween that defines
+  the hero's boundary as it flies from the source to the destination
+  route. During its flight, the hero is moved to an application
+  overlay, so that it appears on top of both routes.
+</div>
+
+<aside class="alert alert-info" markdown="1">
+**Terminology:**
+If the concept of tweens or tweening is new to you, please see the
+[Animations in Flutter tutorial.](/tutorials/animation/)
+</aside>
+
+Hero animations are implemented using two
+[Hero](https://docs.flutter.io/flutter/widgets/Hero-class.html)
+widgets: one describing the widget in the source route,
+and another describing the widget in the destination route.
+From the user’s point of view, the hero appears to be shared, and
+only the programmer needs to understand this implementation detail.
+
+<aside class="alert alert-info" markdown="1">
+**Note about dialogs:**
+Heroes fly from one PageRoute to another. Dialogs
+(displayed with `showDialog()`, for example), use PopupRoutes,
+which are not PageRoutes.  At least for now,
+you can't animate a hero to a Dialog.
+For further developments (and a possible workaround), [watch this
+issue.](https://github.com/flutter/flutter/issues/10667)
+</aside>
+
+Hero animation code has the following structure:
+
+1. Define a starting Hero object, referred to as the _source
+   hero_. The hero specifies its location,
```

Comment:
The hero doesn't specify its size or location, they're just a side effect of laying out the Hero widget.